### ISSUE TITLE
Euclidean loss

### ISSUE THREAD

**Issue by simonMadec** [Neerajj9/CSRNet-keras#9]
I would like to know the reason for the axis=-1 in the euclidean loss computation

```
            def eucli(y_true, y_pred):
                # Euclidean distance as a measure of loss (Loss function) 
                return K.sqrt(K.sum(K.square(y_pred - y_true), axis=-1))
```

My model performs better when I remove this axis (and sum over all the density map)

**Comment by 1286667673** [comment]
hello 
Why my training is wrong? what's the problem
70/700 [==>...........................] - ETA: 1:28:58 - loss: nan - mean_squared_error: nan['data\part_A_final/train_data\images\IMG_137.jpg']
(73, 128, 1)
71/700 [==>...........................] - ETA: 1:28:09 - loss: nan - mean_squared_error: nan['data\part_A_final/train_data\images\IMG_139.jpg']
(46, 68, 1)
72/700 [==>...........................] - ETA: 1:27:07 - loss: nan - mean_squared_error: nan['data\part_A_final/train_data\images\IMG_181.jpg']
loss is nan

**Comment by chajnoven** [comment]
Hello，I have a question for this.
Why My loss is about 0.07 ?
I am not family with the loss and My training process is right? I know I have a problem in 'acc' and it should be 'mse'.

Epoch 1/200
/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:34: DeprecationWarning: Both axis > a.ndim and axis < -a.ndim - 1 are deprecated and will raise an AxisError in the future.
700/700 [==============================] - 1453s 2s/step - loss: 0.0765 - acc: 0.1712
Epoch 2/200
700/700 [==============================] - 597s 853ms/step - loss: 0.0731 - acc: 0.1873
Epoch 3/200
700/700 [==============================] - 510s 728ms/step - loss: 0.0681 - acc: 0.1805
Epoch 4/200
700/700 [==============================] - 500s 715ms/step - loss: 0.0748 - acc: 0.1870
Epoch 5/200
700/700 [==============================] - 501s 716ms/step - loss: 0.0711 - acc: 0.1848
Epoch 6/200
700/700 [==============================] - 500s 715ms/step - loss: 0.0754 - acc: 0.1868
Epoch 7/200
700/700 [==============================] - 498s 712ms/step - loss: 0.0747 - acc: 0.1732
Epoch 8/200
700/700 [==============================] - 493s 705ms/step - loss: 0.0754 - acc: 0.1765
Epoch 9/200
700/700 [==============================] - 495s 707ms/step - loss: 0.0730 - acc: 0.1830
Epoch 10/200
700/700 [==============================] - 507s 724ms/step - loss: 0.0736 - acc: 0.1936
Epoch 11/200
700/700 [==============================] - 464s 662ms/step - loss: 0.0727 - acc: 0.1928
Epoch 12/200
700/700 [==============================] - 455s 651ms/step - loss: 0.0763 - acc: 0.1850
Epoch 13/200
700/700 [==============================] - 456s 652ms/step - loss: 0.0702 - acc: 0.1852

**Comment by Somyarani1113** [comment]
> Hello，I have a question for this.
> Why My loss is about 0.07 ?
> I am not family with the loss and My training process is right? I know I have a problem in 'acc' and it should be 'mse'.
> 
> Epoch 1/200
> /usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:34: DeprecationWarning: Both axis > a.ndim and axis < -a.ndim - 1 are deprecated and will raise an AxisError in the future.
> 700/700 [==============================] - 1453s 2s/step - loss: 0.0765 - acc: 0.1712
> Epoch 2/200
> 700/700 [==============================] - 597s 853ms/step - loss: 0.0731 - acc: 0.1873
> Epoch 3/200
> 700/700 [==============================] - 510s 728ms/step - loss: 0.0681 - acc: 0.1805
> Epoch 4/200
> 700/700 [==============================] - 500s 715ms/step - loss: 0.0748 - acc: 0.1870
> Epoch 5/200
> 700/700 [==============================] - 501s 716ms/step - loss: 0.0711 - acc: 0.1848
> Epoch 6/200
> 700/700 [==============================] - 500s 715ms/step - loss: 0.0754 - acc: 0.1868
> Epoch 7/200
> 700/700 [==============================] - 498s 712ms/step - loss: 0.0747 - acc: 0.1732
> Epoch 8/200
> 700/700 [==============================] - 493s 705ms/step - loss: 0.0754 - acc: 0.1765
> Epoch 9/200
> 700/700 [==============================] - 495s 707ms/step - loss: 0.0730 - acc: 0.1830
> Epoch 10/200
> 700/700 [==============================] - 507s 724ms/step - loss: 0.0736 - acc: 0.1936
> Epoch 11/200
> 700/700 [==============================] - 464s 662ms/step - loss: 0.0727 - acc: 0.1928
> Epoch 12/200
> 700/700 [==============================] - 455s 651ms/step - loss: 0.0763 - acc: 0.1850
> Epoch 13/200
> 700/700 [==============================] - 456s 652ms/step - loss: 0.0702 - acc: 0.1852

Hey can you tell me the optimizer, loss and batch-size used? As I am getting Nan error.
Posting you the error.
1/100 [..............................] - ETA: 12:52 - loss: 0.0838 - mse: 0.0486(683, 1024, 3)
  2/100 [..............................] - ETA: 20:11 - loss: nan - mse: nan      (405, 540, 3)
  3/100 [..............................] - ETA: 19:19 - loss: nan - mse: nan(298, 446, 3)
  4/100 [>.............................] - ETA: 21:17 - loss: nan - mse: nan(298, 446, 3)
  5/100 [>.............................] - ETA: 19:03 - loss: nan - mse: nan(682, 1024, 3)
  6/100 [>.............................] - ETA: 17:07 - loss: nan - mse: nan(768, 1024, 3)
  7/100 [=>............................] - ETA: 16:51 - loss: nan - mse: nan(681, 1024, 3)
  8/100 [=>............................] - ETA: 15:04 - loss: nan - mse: nan(327, 1024, 3)
  9/100 [=>............................] - ETA: 13:40 - loss: nan - mse: nan(282, 377, 3)
 10/100 [==>...........................] - ETA: 13:57 - loss: nan - mse: nan(576, 1024, 3)
 11/100 [==>...........................] - ETA: 14:38 - loss: nan - mse: nan(654, 1024, 3)
 12/100 [==>...........................] - ETA: 13:37 - loss: nan - mse: nan(800, 600, 3)
 13/100 [==>...........................] - ETA: 14:09 - loss: nan - mse: nan(768, 1024, 3)
 14/100 [===>..........................] - ETA: 13:29 - loss: nan - mse: nan(798, 1017, 3)
 15/100 [===>..........................] - ETA: 12:43 - loss: nan - mse: nan(549, 1024, 3)
 16/100 [===>..........................] - ETA: 12:03 - loss: nan - mse: nan(683, 1024, 3)
 17/100 [====>.........................] - ETA: 12:27 - loss: nan - mse: nan(985, 985, 3)
 18/100 [====>.........................] - ETA: 12:55 - loss: nan - mse: nan(551, 1024, 3)
 19/100 [====>.........................] - ETA: 13:13 - loss: nan - mse: nan(450, 600, 3)
 20/100 [=====>........................] - ETA: 12:54 - loss: nan - mse: nan(768, 1024, 3)
 21/100 [=====>........................] - ETA: 12:17 - loss: nan - mse: nan(327, 1024, 3)
 22/100 [=====>........................] - ETA: 12:21 - loss: nan - mse: nan(282, 432, 3)
 23/100 [=====>........................] - ETA: 12:29 - loss: nan - mse: nan(768, 1024, 3)
 24/100 [======>.......................] - ETA: 12:22 - loss: nan - mse: nan(409, 620, 3)
 25/100 [======>.......................] - ETA: 12:33 - loss: nan - mse: nan(600, 900, 3)
 26/100 [======>.......................] - ETA: 12:44 - loss: nan - mse: nan(686, 1024, 3)
 27/100 [=======>......................] - ETA: 12:38 - loss: nan - mse: nan(665, 1024, 3)
 28/100 [=======>......................] - ETA: 12:40 - loss: nan - mse: nan(683, 1024, 3)
 29/100 [=======>......................] - ETA: 12:54 - loss: nan - mse: nan(549, 1024, 3)
 30/100 [========>.....................] - ETA: 12:46 - loss: nan - mse: nan(769, 1024, 3)
 31/100 [========>.....................] - ETA: 12:24 - loss: nan - mse: nan(768, 1024, 3)
 32/100 [========>.....................] - ETA: 12:26 - loss: nan - mse: nan(479, 720, 3)
 33/100 [========>.....................] - ETA: 12:07 - loss: nan - mse: nan(682, 1024, 3)
 34/100 [=========>....................] - ETA: 11:41 - loss: nan - mse: nan(683, 1024, 3)
 35/100 [=========>....................] - ETA: 11:42 - loss: nan - mse: nan(682, 1024, 3)
 36/100 [=========>....................] - ETA: 11:22 - loss: nan - mse: nan(388, 576, 3)
